openfeature-provider/go/state_fetcher.go

```diff
@@ -0,0 +1,180 @@
+package confidence
+
+import (
+	"context"
+	"io"
+	"log"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	adminv1 "github.com/spotify/confidence-resolver-rust/openfeature-provider/go/confidence/proto/confidence/flags/admin/v1"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// FlagsAdminStateFetcher fetches and updates the resolver state from the admin service
+type FlagsAdminStateFetcher struct {
+	resolverStateService adminv1.ResolverStateServiceClient
+	accountName          string
+	etag                 atomic.Value // stores string
+	rawResolverState     atomic.Value // stores []byte
+	resolverStateURI     atomic.Value // stores *adminv1.ResolverStateUriResponse
+	refreshTime          atomic.Value // stores time.Time
+	accountID            string
+	httpClient           *http.Client
+}
+
+// NewFlagsAdminStateFetcher creates a new FlagsAdminStateFetcher
+func NewFlagsAdminStateFetcher(
+	resolverStateService adminv1.ResolverStateServiceClient,
+	accountName string,
+) *FlagsAdminStateFetcher {
+	f := &FlagsAdminStateFetcher{
+		resolverStateService: resolverStateService,
+		accountName:          accountName,
+		httpClient: &http.Client{
+			Timeout: 30 * time.Second,
+		},
+	}
+	// Initialize with empty state
+	emptyState := &adminv1.ResolverState{}
+	if b, err := proto.Marshal(emptyState); err == nil {
+		f.rawResolverState.Store(b)
+	}
+	return f
+}
+
+// RawStateHolder returns the atomic value holding the raw resolver state
+func (f *FlagsAdminStateFetcher) RawStateHolder() *atomic.Value {
+	return &f.rawResolverState
+}
```

Comment:
nit: Let's delete this unused function, so we don't break encapsulation and risk external manipulation of this value in the future